### PR TITLE
Implement no-double-backprop version of F.softmax_cross_entropy using FunctionNode

### PR DESCRIPTION
**Pull request by niboshi**
Implements `SoftmaxCrossEntropy` with `FunctionNode`.

This execution path is for single-backward only, so double-backward is not implemented.
This PR does not enhance functionality, but the overhead of `FunctionAdapter` could be reduced.

### CODE REVIEW

File: chainer/functions/loss/softmax_cross_entropy.py [hunk 1]
```diff
@@ -250,6 +269,11 @@ def backward_gpu(self, inputs, grad_outputs):
 
         return gx, None
 
+    def backward(self, input_indexes, grad_outputs):
+        assert False, (
+            'This execution path of F.softmax_cross_entropy does not support '
+            'double-backprop. There must be a bug.')
```
Comment (beam2d) [inline chainer/functions/loss/softmax_cross_entropy.py]: Better to advise the user to use `enable_double_backprop=True`.

**Comment by niboshi** [comment]
PTAL

**Comment by beam2d** [comment]
Jenkins, test this please.

**Comment by beam2d** [comment]
LGTM
